scripts/backtesting_agent.py

```
@@ -0,0 +1,36 @@
+from .db_connections import ConnectionManager
+from .trading_db import list_strategies, record_backtest
+
+
+def run_backtest(prices: pd.DataFrame) -> dict:
+    """Simple backtest: buy and hold return."""
+    if prices.empty:
+        return {"return": 0.0}
+
+    start = prices["close"].iloc[0]
+    if start == 0:
+        return {"return": float('inf') if end > 0 else 0.0}
```

Ruff (0.11.9)

Undefined name `end`

(F821)